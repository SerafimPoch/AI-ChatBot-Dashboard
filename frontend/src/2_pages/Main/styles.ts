import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import { LAPTOP_MQ, TABLET_MQ } from "@/src/6_shared/media-queries";
import styled from "@emotion/styled";

export const MainContainer = styled.main`
  ${() => {
    const themeColors = useThemeContext();

    return `
      width: 100vw;
      padding: 20px 64px 20px 20px;
      background: ${themeColors?.coreBackground};
      display: flex;
      justify-content: center;
      gap: 64px;
      align-items: stretch;

      @media(max-width:${LAPTOP_MQ}px) {
        gap: 40px;
      }

      @media(max-width:${TABLET_MQ}px) {
        gap: 20px;
        padding: 20px;
      }
    `;
  }}
`;

export const Container = styled.div`
  width: 100%;
  max-width: 1440px;
`;

export const MainContentWrapper = styled.div`
  display: grid;
  grid-template-columns: minmax(700px, 1.5fr) minmax(500px, 1fr);
  grid-row-gap: 8px;
  grid-column-gap: 8px;

  @media (max-width: ${LAPTOP_MQ}px) {
    grid-template-columns: 1fr 1fr;
  }

  @media (max-width: ${TABLET_MQ}px) {
    grid-template-columns: 1fr;
  }
`;
