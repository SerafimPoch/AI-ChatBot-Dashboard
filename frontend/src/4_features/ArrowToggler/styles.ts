import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import styled from "@emotion/styled";

export const ArrowTogglerContainer = styled.div`
  ${() => {
    const themeColors = useThemeContext();

    return `
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      gap: 32px;
      align-self: stretch;
      border-radius: 12px;
      background: ${themeColors?.grey};
      width: 56px;
    `;
  }}
`;
