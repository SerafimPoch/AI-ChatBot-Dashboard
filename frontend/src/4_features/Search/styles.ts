import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import styled from "@emotion/styled";

export const SearchContainer = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  gap: 8px;
`;

export const SearchInput = styled.input`
  ${() => {
    const themeColors = useThemeContext();

    return `
      background: transparent;
      border: 1px solid #5d74f1;
      outline: none;
      border-radius: 6px;
      padding: 0 5px;
      color: ${themeColors?.textPrimary};
    `;
  }}
`;
