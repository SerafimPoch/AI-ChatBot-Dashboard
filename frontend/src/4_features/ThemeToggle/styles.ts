import styled from "@emotion/styled";

interface IThemeItem {
  active: boolean;
}

export const ThemeToggleContainer = styled.div`
  display: flex;
  align-items: center;
  border-radius: 8px;
  border: 2px solid #f2f2f2;
  width: 100%;
  height: 40px;
`;

export const ThemeItem = styled.button<IThemeItem>`
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 6px;
  width: 50%;
  height: 32px;
  background: ${({ active }) => (active ? "#f2f2f2" : "transparent")};
  color: ${({ active }) =>
    active ? "var(--text-primary)" : "var(--text-secondary)"};
  margin: 0 4px;
  transition: background-color 0.3s ease, color 0.3s ease;
`;
