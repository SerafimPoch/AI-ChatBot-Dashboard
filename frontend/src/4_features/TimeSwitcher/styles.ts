import styled from "@emotion/styled";

interface TTimeSwitcherItem {
  active: boolean;
}

export const TimeSwitcherContainer = styled.div`
  display: flex;
  align-items: center;
  gap: 16px;
`;

export const TimeSwitcherItem = styled.button<TTimeSwitcherItem>`
  width: fit-content;
  min-width: 33px;
  height: 28px;
  padding: 4px 8px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
  font-family: var(--font-sf-pro-rounded);
  color: ${({ active }) => (active ? "#fff" : "#A0A5A9")};
  background-color: ${({ active }) => (active ? "#1D1D1D" : "transparent")};
  border-radius: 8px;
  transition: background-color 0.3s ease, color 0.3s ease;

  &:hover {
    color: #fff;
    background-color: #1d1d1d;
  }
`;
