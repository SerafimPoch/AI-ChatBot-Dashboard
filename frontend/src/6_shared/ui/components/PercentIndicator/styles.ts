import styled from "@emotion/styled";

interface TPercentIndicatorValue {
  isPercentNegative: boolean;
}

export const PercentIndicatorContainer = styled.div`
  display: flex;
  align-items: center;
  gap: 2px;
`;

export const PercentIndicatorTriangle = styled.div<TPercentIndicatorValue>`
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-bottom-style: solid;
  border-bottom-width: 10px;
  border-bottom-color: ${({ isPercentNegative }) =>
    isPercentNegative ? "#EA7465" : "#349264"};
  rotate: ${({ isPercentNegative }) => (isPercentNegative ? "180deg" : "none")};
`;

export const PercentIndicatorValue = styled.p<TPercentIndicatorValue>`
  font-family: var(--font-sf-pro-rounded);
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
  color: ${({ isPercentNegative }) =>
    isPercentNegative ? "#EA7465" : "#349264"};
`;
