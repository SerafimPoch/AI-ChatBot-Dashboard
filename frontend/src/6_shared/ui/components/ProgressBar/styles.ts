import styled from "@emotion/styled";

interface TProgressBarLine {
  width: number;
  color: string;
}

export const ProgressBarContainer = styled.div`
  height: 24px;
  border-radius: 4px;
  background: #f2f2f2;
`;

export const ProgressBarLine = styled.div<TProgressBarLine>`
  width: ${({ width }) => `${width}%`};
  display: flex;
  align-items: center;
  gap: 8px;
  border-radius: 4px;
  background: ${({ color }) => color};
  transition: width 0.4s ease;
  height: 100%;
  padding: 0 8px;
`;
