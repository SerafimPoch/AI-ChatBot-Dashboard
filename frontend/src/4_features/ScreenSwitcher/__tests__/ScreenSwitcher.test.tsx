import React from "react";
import { render } from "@testing-library/react";
import { ScreenSwitcherContainer } from "../styles";

describe("ScreenSwitcher page components", () => {
  it("LogoutContainer renders correctly", () => {
    const { asFragment } = render(<ScreenSwitcherContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
