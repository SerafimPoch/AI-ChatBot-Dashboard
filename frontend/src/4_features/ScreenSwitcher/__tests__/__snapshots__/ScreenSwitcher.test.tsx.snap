// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenSwitcher page components LogoutContainer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 24px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;
